File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListView/Performance/PagingPage.vb
Namespace CompanionKit.Controls.ListView.Performance

	Public Class PagingPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

            ' Fill up ListView
			Dim i As Integer
			For i = 1 To 30
                Me.listViewWithPaging.Items.Add(New ListViewItem(New String() {String.Format("User{0}", i), String.Format("user{0}@gmail.com", i)}))
                Me.listViewWithoutPaging.Items.Add(New ListViewItem(New String() {String.Format("User{0}", i), String.Format("user{0}@gmail.com", i)}))
			Next i

            Me.itemsPerPagesNumUpDown.Value = Me.listViewWithPaging.ItemsPerPage

		End Sub
        ''' <summary>
        ''' Handles the ValueChanged event of the itemsPerPagesNumUpDown control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub itemsPerPagesNumUpDown_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles itemsPerPagesNumUpDown.ValueChanged
            Me.listViewWithPaging.ItemsPerPage = Decimal.ToInt32(Me.itemsPerPagesNumUpDown.Value)
        End Sub
	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListView/Performance/PagingPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.ListView.Performance
{
    public partial class PagingPage : UserControl
    {
        public PagingPage()
        {
            InitializeComponent();
            
            // Fill up ListView
            for (int i = 1; i < 31; ++i)
            {
                this.listViewWithPaging.Items.Add(new ListViewItem(new string[] { string.Format("User{0}", i), string.Format("user{0}@gmail.com", i) }));
                this.listViewWithoutPaging.Items.Add(new ListViewItem(new string[] { string.Format("User{0}", i), string.Format("user{0}@gmail.com", i) }));
            }

            this.itemsPerPagesNumUpDown.Value = this.listViewWithPaging.ItemsPerPage;
            }

        /// <summary>
        /// Handles the ValueChanged event of the itemsPerPagesNumUpDown control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void itemsPerPagesNumUpDown_ValueChanged(object sender, EventArgs e)
        {
            this.listViewWithPaging.ItemsPerPage = decimal.ToInt32(this.itemsPerPagesNumUpDown.Value);
        }

       
    }
}